README.md

```diff
@@ -1,32 +1,506 @@
-# Lesson Title
+# Nested Routing
 
 ## Learning Goals
 
-- Learning Goal 1
-- Learning Goal 2
+- Created nested routes using `children` and `Outlet`
+- DRY up code with nested routing
+- Pass data to nested route components using `useOutletContext`
 
 ## Introduction
 
-A 1-2 sentence summary of what will be covered.
+In this code along, we're going to keep working with our Social Media
+application we made in the previous code along. However, we want to make some
+updates!
 
-## Topic 1
+First of all, we don't want to have to include our `NavBar` component in every
+page level component - that wasn't very DRY! We also included the same ErrorPage
+on every one of our components - we'll fix that too.
 
-Discuss the topic of the reading. Break into sections and sub-sections as
-appropriate to help with the reader's orientation with the material and future
-referencing.
+Second of all, we don't want to navigate to a brand new web page to view a
+specific user. Instead, we want that user to display on the same page as the
+list of users! But, we do still want each user to have it's own URL associated
+with it, so that we can share links to specific users if we want to.
 
-## Topic 2
+All of this can be done using **Nested Routing**. Nested Routing allows us to
+re-render specific _pieces_ of a webpage when a user navigates to a new route,
+rather than re-rendering the entire page. This can be great for developers, as
+it allows easy re-use of certain components, as well as for users, as it can
+make navigating a website smoother and easier.
 
-Discuss the topic of the reading. Break into sections and sub-sections as
-appropriate to help with the reader's orientation with the material and future
-referencing.
+To add Nested Routing to our application, we'll need to use a few other things
+from `react-router-dom` - the `children` attribute on our route objects, the
+`Outlet` component, and the `useOutletContext` hook. Let's dive into each of
+those in turn!
+
+## Rendering Nested Routes as "children"
+
+In our last code along, you might have noticed that we didn't have an `App`
+component in our list of components. In fact, there was no single parent
+component to our whole application! Instead, we just had a bunch of parallel
+components, each of which was rendering on its own route.
+
+While this parallel approach definitely works, and might be the right decision
+depending on the app you're building, it has some drawbacks. As mentioned above,
+we had some code that wasn't very DRY - we used the NavBar component in every
+one of our page views, and gave each of our routes the same exact errorElement.
+
+Moreover, the only way we could have declared global state for our application
+would have been through creating our own contextProvider with the `useContext`
```

Comment:
Good idea! I was thinking about linking to the previous Canvas lesson on the topic. Is there a best practices way to do that so that it works well with different Canvas instances? Or would it be better to link to outside documentation?